Source/WebCore/css/html.css

```diff
@@ -319,6 +319,13 @@ tr {
     border-color: inherit;
 }
 
+/* https://html.spec.whatwg.org/#tables-2 */
+/* When the td/th are inside a table (the normal case), the padding is taken
+   care of by HTMLTableCellElement::additionalPresentationAttributeStyle(). */
```

Comment:
This function does not exist. Do you know where this actually happens?

Reply:
https://searchfox.org/wubkat/source/Source/WebCore/html/HTMLTableCellElement.cpp#139

Reply:
`HTMLTableCellElement::additionalPresentationalHintStyle()`

Reply:
There's a slight difference in name in your comment fwiw `Attribute` vs `Hint`

Reply:
That being said, the Chromium approach feels like cheating for the sake of the WPT. Would be good to understand why WebKit only adds conditionally 1px padding when in a table. Is this something the spec says? If not, the rule should probably be exactly the rule in the spec: `td, th { padding: 1px; }` and we should remove these lines: https://searchfox.org/wubkat/rev/02b2f8b7a253140d82f59e38a68257431139d2e7/Source/WebCore/html/HTMLTableElement.cpp#519,524

Reply:
Web-Spec: _"If the attribute is present but parsing the attribute's value using the [rules for parsing non-negative integers](https://html.spec.whatwg.org/#rules-for-parsing-non-negative-integers) generates an error, a default value of 1px is expected to be used for that property instead."_

Reply:
That's for `<table>` itself not `<td>` or `<th>`, unless I'm looking at the wrong line.

Reply:
Looking into CSS Table Layout spec: https://drafts.csswg.org/css-tables-3/#fixup & https://drafts.csswg.org/css-tables-3/#mapping

Reply:
This comment from Chrome bug gives more context: https://bugs.chromium.org/p/chromium/issues/detail?id=1406929#c6

Reply:
So in summary, what I understand overall: for normal syntax with `td` etc. in `table` element, we have C++ code to add 'cellpadding' of '1px' and handle accordingly as per web-spec but when we don't follow syntax and outside of `table`, Blink modified UA spec rule to accommodate and manage it in UA stylesheet.